enumerable.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each(arr)
+    x = 0
+    while x < arr.length
+      yield(arr[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index(arr)
+    i = 0
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  def my_select(arr)
+    y = 0
+    new_arr = []
+    while y < arr.length
+      new_arr.push(arr[y]) if yield(arr[y])
+      y += 1
+    end
+    new_arr
+  end
+
+  def my_all?(arr)
+    w = 0
+    while w < arr.length
+      return false unless yield(arr[w])
+
+      w += 1
+    end
+    true
+  end
+
+  def any?(arr)
+    i = 0
+    while i < arr.length
+      return true if yield(arr[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?(arr)
+    i = 0
+    while i < arr.length
+      return false if yield(arr[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(arr)
+    i = 0
+    results = 0
+    while i < arr.length
+      results += 1 if yield(arr[i])
+      i += 1
+    end
+    results
+  end
```

Comment:
- [ ] Enumerables `my_count` returns the number of items in enum through enumeration
`[1,2,3,4,5].my_count` should return `5`

- [ ] Enumerables `my_count` counts the number of items in enum that are equal to the item if an argument is given
`[1,2,3,4,5].my_count(3)` should return `1`

- [ ] Enumerables `my_count` counts the number of elements yielding a true value if a block is given
`[1,2,3,4,5].my_count{ |num| num == 4 }` should return `1`